src/template.js

```diff
@@ -35,12 +44,37 @@ switch (data.method) {
       break;
     }
 
-    if (queryPermission('inject_script', INSIGHTS_LIBRARY_URL)) {
+    const url = getLibraryURL(data.useIIFE);
+
+    if (queryPermission('inject_script', url)) {
       injectScript(
-        INSIGHTS_LIBRARY_URL,
-        data.gtmOnSuccess,
+        url,
+        () => {
+          if (!copyFromWindow('aa')) {
+            data.gtmOnFailure();
+            logger('[ERROR] Failed to load search-insights.');
+            return;
+          }
+
+          let libraryLoaded = false;
+          aa('getUserToken', null, () => {
```

Comment:
If the library is fully loaded, this method should work.

Reply:
It might be worth adding a comment explaining that aa('getUserToken') is synchronous so the flag will be set immediately.